### PR TITLE
Sm/circle promotions

### PR DESCRIPTION
**Pull request by mshanemc**
### What does this PR do?
Every Thursday morning, kick off a job that waits for approval to 
- open a CTC case
- promote installers based on npm and github tags to the stable channel (you can override VERSION/SHA in env if the tags are messed up)
- promote npm packages from latest-rc to latest
- complete the CTC steps

### What issues does this PR fix or reference?
@W-7869694 (CTC)
@W-9032069@ (npm promotes)
@W-9032027@ artifacts on stable channel

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -270,12 +271,15 @@ workflows:
   promote-to-latest:
     triggers:
       - schedule:
-          # Thursdays
-          cron: 27 15 * * 4
+          # Thursday morning
+          cron: 0 11 * * 4
+          # testing
```
Comment (peternhale) [inline .circleci/config.yml:276]: @mshanemc please remove

File: .circleci/config.yml [hunk 2]
```diff
@@ -270,12 +271,15 @@ workflows:
...
+          # cron: 26 23 * * 4
           filters:
             branches:
               only:
                 - main
+                # - sm/circle-promotions
```
Comment (peternhale) [inline .circleci/config.yml:282]: Ditto

File: scripts/promote [hunk 3]
```diff
@@ -2,8 +2,10 @@
 
 set -ex
 
+CHANNEL=latest
+
 # test channel
-CHANNEL=latest-ci-test
+# CHANNEL=latest-ci-test
```
Comment (peternhale) [inline scripts/promote:8]: Ditto

**Review by peternhale — APPROVED** [review]
@mshanemc just nits

**Review by peternhale — APPROVED** [review]

File: scripts/dist-tags-promote [hunk 4]
```diff
@@ -1,5 +1,14 @@
+# get CI environment auth'd; local one-off use assumes you're already auth'd to npm
+if [ $NPM_TOKEN ]; then
+  echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > .npmrc
+fi
+
 CLI_RC_VERSION=$(npm show sfdx-cli --json | jq -r '.["dist-tags"] | .["latest-rc"]')
+# use the following line for testing without making changes
+# npm dist-tag add sfdx-cli@${CLI_RC_VERSION} latest-rc
```
Comment (RodEsp) [inline scripts/dist-tags-promote:8]: What are these lines for? Are they really necessary?

File: scripts/dist-tags-promote [hunk 5]
```diff
@@ -1,5 +1,14 @@
...
 npm dist-tag add sfdx-cli@${CLI_RC_VERSION} latest
 
 DX_RC_VERSION=$(npm show salesforcedx --json | jq -r '.["dist-tags"] | .["latest-rc"]')
+# use the following line for testing without making changes
+# npm dist-tag add sfdx-cli@${CLI_RC_VERSION} latest-rc
```
Comment (RodEsp) [inline scripts/dist-tags-promote:13]: What are these lines for? Are they really necessary?

**Review by maggiben — APPROVED** [review]

File: scripts/dist-tags-promote [hunk 6]
```diff
@@ -1,5 +1,14 @@
+# get CI environment auth'd; local one-off use assumes you're already auth'd to npm
+if [ $NPM_TOKEN ]; then
+  echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > .npmrc
+fi
+
 CLI_RC_VERSION=$(npm show sfdx-cli --json | jq -r '.["dist-tags"] | .["latest-rc"]')
```
Comment (amphro) [inline scripts/dist-tags-promote:6]: If you want, you can just do `npm show sfdx-cli@latest-rc version`

**Review by amphro — APPROVED** [review]

File: scripts/dist-tags-promote (hunk 6 above)
Comment (mshanemc) [reply scripts/dist-tags-promote]: yeah, that's so much better.

File: scripts/dist-tags-promote (hunk 4 above)
Comment (mshanemc) [reply scripts/dist-tags-promote]: easy way to replicate the "dry run" feature we had before.  Removing.

**Pull request merged**